src/stubs/reactive_stubs_support.h

```diff
@@ -46,7 +46,6 @@ extern char __sm_num_inputs;
 #define SM_NAME X
 
 // declare symbols for the public/secret regions
-extern void *__sm_X_public_start, *__sm_X_public_end,
-            *__sm_X_secret_start, *__sm_X_secret_end;
+extern char __PS(SM_NAME), __PE(SM_NAME), __SS(SM_NAME), __SE(SM_NAME);
```

Comment:
Hey @jovanbulck, using the macros here makes a lot of sense, but using `SM_NAME` instead of `X` does not work, because the library does not compile. As pointed [in this post](https://stackoverflow.com/a/31294244), the problem seems that the macro `SM_NAME` is expanded _after_ the macros `__PS` etc, not before.

Following the suggestion in that post, we could add another macro level, like this:
```c
#define SM_NAME X
#define ___PS(name) __PS(name)
#define ___PE(name) __PE(name)
#define ___SS(name) __SS(name)
#define ___SE(name) __SE(name)


// declare symbols for the public/secret regions
extern char ___PS(SM_NAME), ___PE(SM_NAME), ___SS(SM_NAME), ___SE(SM_NAME);
```

As an alternative, we could simply replace `SM_NAME` with `X`. Or maybe you will come up with a neater solution :)

Reply:
Something like this also works:

```c
#define SM_NAME X
#define __SECTION(inner, name) inner(name)

// declare symbols for the public/secret regions
extern char __SECTION(__PS, SM_NAME), __SECTION(__PE, SM_NAME), __SECTION(__SS, SM_NAME), __SECTION(__SE, SM_NAME);
```

Reply:
oh right, good catch, nesting C macros is tricky indeed :sweat_smile: 

I think your solution is a good workaround indeed